qiskit_experiments/library/characterization/cr_hamiltonian_analysis.py

```diff
@@ -0,0 +1,294 @@
+                p_{z, c} p_{y, c} - p_{z, c} p_{y, c} \cos(\Omega_c t) -
+                \Omega_c p_{x, c} \sin(\Omega_c t) \right) + b \ ... \ (2), \\
+            F_{z, c}(t) &= \frac{1}{\Omega_c^2} \left(
+             p_{z, c}^2 + (p_{x, c}^2 + p_{y, c}^2) \cos(\Omega_c t) \right) + b \ ... \ (3),
+
+        where :math:`\Omega_c = \sqrt{p_{x, c}^2+p_{y, c}^2+p_{z, c}^2}` and
+        :math:`p_{x, c}, p_{y, c}, p_{z, c}, b` are the fit parameters.
+        The subscript :math:`c` represents the state of control qubit :math:`c \in \{0, 1\}`.
+        The fit functions :math:`F_{x, c}, F_{y, c}, F_{z, c}` approximate the Pauli expectation
+        value of target qubit :math:`\langle \sigma_{x, c} (t) \rangle,
+        \langle \sigma_{y, c} (t) \rangle, \langle \sigma_{z, c} (t) \rangle`, respectively.
+
+        Based on the fit result, cross resonance Hamiltonian coefficients can be written as
+
+        .. math::
+
+            ZX &= \frac{p_{x, 0} - p_{x, 1}}{2} \\
+            ZY &= \frac{p_{y, 0} - p_{y, 1}}{2} \\
+            ZZ &= \frac{p_{z, 0} - p_{z, 1}}{2} \\
+            IX &= \frac{p_{x, 0} + p_{x, 1}}{2} \\
+            IY &= \frac{p_{y, 0} + p_{y, 1}}{2} \\
+            IZ &= \frac{p_{z, 0} + p_{z, 1}}{2}
+
+        In this analysis, initial guess is generated by the following equations.
+
+        .. math::
+
+            p_x &= \omega \cos(\theta) \cos(\phi) \\
+            p_y &= \omega \cos(\theta) \sin(\phi) \\
+            p_z &= \omega \sin(\theta)
+
+        where $\omega$ is the mean oscillation frequency of eigenvalues,
+        $\theta = \cos^{-1}\sqrt{\frac{\max F_z - \min F_z}{2}}$ and $\phi \in [-\pi, \pi]$.
+
+    # section: fit_parameters
+
+        defpar p_{x, 0}:
+            desc: Fit parameter of oscillations when control qubit state is 0.
+            init_guess: See fit model section.
+            bounds: None
+
+        defpar p_{y, 0}:
+            desc: Fit parameter of oscillations when control qubit state is 0.
+            init_guess: See fit model section.
+            bounds: None
+
+        defpar p_{z, 0}:
+            desc: Fit parameter of oscillations when control qubit state is 0.
+            init_guess: See fit model section.
+            bounds: None
+
+        defpar p_{x, 1}:
+            desc: Fit parameter of oscillations when control qubit state is 1.
+            init_guess: See fit model section.
+            bounds: None
+
+        defpar p_{y, 1}:
+            desc: Fit parameter of oscillations when control qubit state is 1.
+            init_guess: See fit model section.
+            bounds: None
+
+        defpar p_{z, 1}:
+            desc: Fit parameter of oscillations when control qubit state is 1.
+            init_guess: See fit model section.
+            bounds: None
+
+    """
+
+    __series__ = [
+        curve.SeriesDef(
+            name="x|c=0",
+            fit_func=lambda x, px0, px1, py0, py1, pz0, pz1, b: curve.fit_function.bloch_oscillation_x(
```

Comment:
Is the curve fit able to figure out the parameter dependence fairly well (i.e. that the data is independent of three of the parameters for each of the series)? You could fit the data for the two control states separately though I don't know if our curve fitting code supports that.

Reply:
I think so. Each subset of data with different control qubit state is used to fit different set of parameters. All of these curves are simultaneously fit so the dependency should be implicitly distinguished by the fit algorithm. Unfortunately we don't have mechanism to call the fitter multiple times with different subset of data. To realize this, we need to define the ham tomo as a batch experiment and combine the result afterwards.